### PR TITLE
Request logging

### PR DESCRIPTION
**Pull request by KayEss**
Adds new middleware that records information about processed requests to a Postgres database.

### CODE REVIEW

File: Extras/request-logger/request-logger.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+CREATE TABLE request_log (
```
Comment (tleekkul) [inline Extras/request-logger/request-logger.sql]: We should follow the directory structure and name convention of either `odin` or `tormor`. So we can use those to handle the migrate.

Like `/Schema/request-logger/001-initial.blue.sql`

File: Extras/request-logger/logger.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                    fostlib::insert(row, "status", response.second);
+                } catch (std::exception const &e) {
+                    exception = std::current_exception();
+                    fostlib::insert(row, "exception", e.what());
+                } catch (...) {
+                    exception = std::current_exception();
+                    fostlib::insert(row, "exception", "**unknown**");
+                }
+            }
+            fostlib::insert(
```
Comment (tleekkul) [inline Extras/request-logger/logger.cpp]: missing log flush, just to make that we get all of them.

File: Extras/request-logger/request-logger.fg [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        "exception": null,
+        "status": 200,
+        "response_headers": {
+            "Content-Length": "67",
+            "Content-Transfer-Encoding" : "8bit",
+            "Content-Type" : "text/html; charset=\"utf-8\""
+        }
+    }
+
+setting webserver views/rq-log {
```
Comment (tleekkul) [inline Extras/request-logger/request-logger.fg]: Add the note to state that this is not the recommended configuration

**Comment by KayEss** [comment]
I discussed this with Sajal. He wants to make sure that a failure in the logging database doesn't cause an outage in production traffic, so going to refactor the timing of when things happen a bit and ignore any exceptions thrown during log recording.

**Pull request merged**